internal/util/util.go

```diff
@@ -311,6 +311,13 @@ func GetVolumeSnapshotbackupWithStatusData(volumeSnapshotbackupNS string, volume
 		}
 
 		if len(vsb.Status.Conditions) == 0 {
+			if vsb.Status.Phase == datamoverv1alpha1.SnapMoverBackupPhaseFailed ||
+				vsb.Status.Phase == datamoverv1alpha1.SnapMoverBackupPhasePartiallyFailed {
+				return false, errors.Errorf("volumesnapshotbackup %v has failed status", vsb.Name)
+			}
+			if vsb.Status.Phase == datamoverv1alpha1.SnapMoverBackupPhaseCompleted {
+				return false, errors.Errorf("volumesnapshotbackup %v completed without conditions", vsb.Name)
```

Comment:
Why this is also considered `failed` condition with error?

Reply:
Because there are no conditions.. in the following few lines if you expand you'll note that this code expect there to be a condition prior to returning true.

Reply:
Does it mean the backup failed / completed or it's really in some wired unknown state?